torch/csrc/jit/script/module.h

```diff
@@ -105,124 +122,133 @@ struct NamedParameter {
   std::unique_ptr<at::Tensor> parameter;
 };
 
-struct NamedMember {
-  enum Kind { Module, Parameter, Method, None };
-  // note: None is used to report undefined attributes;
-  Kind kind;
-  size_t offset;
-
-  static const char * kind_string(Kind kind) {
-    switch(kind) {
-      case Module: return "module";
-      case Parameter: return "parameter";
-      case Method: return "method";
-      case None: return "none";
-      default: return "unknown";
+// simple ordered dict used only in Module
+// contains only the minimum necessary functionality for Module
+template<typename T>
+struct OrderedDict {
+  OrderedDict() {}
+  T& insert(const std::string& name,  T&& value, const char* what) {
+    if(index_.count(name) != 0) {
```

Comment:
FWIW `OrderedDict`s support insertion (items stay in the same order, only value is updated). Don't know if this is a general constraint coming from Modules, or you didn't implement it for some other reason